src/lib/Zikula/Core/Theme/AssetBag.php

```diff
@@ -19,18 +39,22 @@ public function __construct(array $assets = array())
      */
     public function add($asset)
     {
+        // ensure value is an array
         if (!is_array($asset)) {
-            $asset = array($asset);
+            $asset = [$asset => self::DEFAULT_WEIGHT];
         }
-        $this->assets = array_merge($this->assets, $asset);
-        $this->assets = array_unique($this->assets);
+        foreach ($asset as $source => $weight) {
+            if ((isset($this->assets[$source]) && $this->assets[$source] > $weight) || !isset($this->assets[$source])) {
+                // keep original weight if lighter. set if not set already.
+                $this->assets[$source] = $weight;
+            }
+        }
+        asort($this->assets); // put array in order by weight
```

Comment:
This is a BC break then. Order matters almost all the time, because I often include a third party js library in one of my modules and use it in a javascript file.
```twig
    {{ pageAddVar('javascript', zasset('@MyModule:js/vendor/toastr.min.js')) }}
    {{ pageAddVar('javascript', zasset('@MyModule:js/use-toastr.js')) }}
```